lib/decorators/on-events.decorator.ts

```diff
@@ -0,0 +1,26 @@
+import { OnEventOptions } from '../interfaces';
+import { OnEventMetadata, OnEventType } from './on-event.decorator';
+import { extendArrayMetadata } from '@nestjs/common/utils/extend-metadata.util';
+import { EVENT_LISTENER_METADATA } from '../constants';
+
+/**
+ * Event listener decorator.
+ * Subscribes to array of events based on the specified name(s).
+ *
+ * @param events array of events to subscribe to
+ */
+export const OnEvents = (
```

Comment:
Instead of introducing a new decorator, could we just modify the `@OnEvent` implementation to use the "extendArrayMetadata" under the hood?

Reply:
The original decorator already accepts string|string[], so introducing string[]|string[][] would not be backwards compatible (can't tell if the string array is an array of events, or a single event with array representing the namespaces)

Reply:
I'm not saying that we should update the decorator signature, but just use the "extendArrayMetadata" function under the hood, as you did in the `OnEvents` decorator

Reply:
Without OnEvents decorator it would still be impossible to attach events based on a dynamic array (say, based on a config class etc)

Reply:
That's fine, we don't want to support such a feature in this package

Reply:
Then I can remove this, but this would force users who need this use-case (incl me and commenters on the original issue) to resort to monkey-patching decorator metadata with custom code instead of an easy to use decorator.

Reply:
If you want to dynamically retrieve event names (and dynamically register handlers for them), you shouldn't use decorators in the first place

Reply:
Example of dynamic parameters in decorators from nestjs docs https://docs.nestjs.com/fundamentals/custom-providers#class-providers-useclass

---

Removing `OnEvents` from PR.

Reply:
This is a different use case than what's stated in the docs

Reply:
Removed